database/engine/rrdengine.c

```diff
@@ -503,15 +608,24 @@ static void after_delete_old_data(uv_work_t *req, int status)
     ctx->disk_space -= deleted_bytes;
     info("Reclaimed %u bytes of disk space.", deleted_bytes);
 
+    error = uv_thread_join(wc->now_deleting_files);
+    if (error) {
+        error("uv_thread_join(): %s", uv_strerror(error));
+    }
+    freez(wc->now_deleting_files);
     /* unfreeze command processing */
-    wc->now_deleting.data = NULL;
-    /* wake up event loop */
-    assert(0 == uv_async_send(&wc->async));
+    wc->now_deleting_files = NULL;
+
+    wc->cleanup_thread_deleting_files = 0;
+
+    /* interrupt event loop */
+    uv_stop(wc->loop);
 }
 
-static void delete_old_data(uv_work_t *req)
+static void delete_old_data(void *arg)
 {
-    struct rrdengine_instance *ctx = req->data;
+    struct rrdengine_instance *ctx = arg;
```

Comment:
With gcc 9.2 I did not have warnings or errors to compile the code, but should we not add a explicit cast to `struct rrdengine_instance *` here to avoid warnings with old compilers?

Reply:
It's one of the differences between C and C++, there should be no warning here.

https://stackoverflow.com/questions/1736833/void-pointers-difference-between-c-and-c

Reply:
All right.